### DISCUSSION TITLE
新章节如何不换页

### DISCUSSION THREAD

**Discussion by karrymao** [sjtug/SJTUThesis#731] [提问]
开始新的一章，默认会从新的一页开始排版，如何使其连续？

**Comment by skyzh** [comment]
看起来没有办法简单修改。可以考虑把模版的这一行去掉：

https://github.com/sjtug/SJTUThesis/blob/90ce4a3d5fb729143bc57c8e2db021417a985f90/texmf/tex/latex/sjtuthesis/sjtuthesis.cls#L808

**Reply by karrymao** [reply]
> 看起来没有办法简单修改。可以考虑把模版的这一行去掉：
> 
> https://github.com/sjtug/SJTUThesis/blob/90ce4a3d5fb729143bc57c8e2db021417a985f90/texmf/tex/latex/sjtuthesis/sjtuthesis.cls#L808

在Overleaf上没有效果。

**Reply by skyzh** [reply]
https://liam.page/2015/04/22/cancle-clearpage-in-chapter/ 参考这个打补丁吧，看起来是 LaTeX 的默认行为。

**Reply by karrymao** [reply]
> https://liam.page/2015/04/22/cancle-clearpage-in-chapter/ 参考这个打补丁吧，看起来是 LaTeX 的默认行为。

尝试了使用xpatch包进行替换和直接修改sjtuthesis.cls中chapter定义的第一行，都没有效果。

**Comment by LogCreative** [comment]
简而言之，直接将换页操作局部地定义为空操作。这里也重定义 `\cleardoublepage` 以防使用了 `openright` 文档类设置。

```latex
{
\renewcommand\clearpage{}
\renewcommand\cleardoublepage{}
\begin{abstract}

\end{abstract}
}
```
这里的 abstract 环境也可以换成 chapter 命令。

<details>
<summary>
另解
</summary>

abstract 不仅仅只是一个 chapter，翻阅[定义](https://github.com/sjtug/SJTUThesis/blob/700093d999dc6fe212aa280d04aee6c120470f3b/texmf/tex/latex/sjtuthesis/sjtuthesis-undergraduate.ltx#L71-L72)可以看到 abstract 用了内部宏 `\sjtu@chapter` 包装（acknowledgement 也是[一样的](https://github.com/sjtug/SJTUThesis/blob/700093d999dc6fe212aa280d04aee6c120470f3b/texmf/tex/latex/sjtuthesis/sjtuthesis.cls#L971-L978)），而这个宏的[定义](https://github.com/sjtug/SJTUThesis/blob/700093d999dc6fe212aa280d04aee6c120470f3b/texmf/tex/latex/sjtuthesis/sjtuthesis.cls#L801-L813)开头就是

https://github.com/sjtug/SJTUThesis/blob/700093d999dc6fe212aa280d04aee6c120470f3b/texmf/tex/latex/sjtuthesis/sjtuthesis.cls#L802

也就是 chapter/break 的原始定义，也就意味着 `\clearpage` 除了 chapter 会产生，使用` \sjtu@chapter` 也会产生。由于多个 `\clearpage` 的效果等效为一个，所以只会断一次页。

所以另外一种解法是
```latex
\ctexset{chapter/break={}} % <-- 使 chapter 不换页
\makeatletter
\RenewDocumentCommand{\sjtu@chapter}{s O{#3} m O{#2}}{
  % \if@openright\cleardoublepage\else\clearpage\fi% <-- 使之无效
  \IfBooleanTF{#1}{%
    \ifthenelse{\equal{#4}{}}{}{%
      \sjtu@pdfbookmark{0}{#4}%
    }
  }{%
    \sjtu@phantomsection
    \addcontentsline{toc}{chapter}{#4}%
  }%%
  \chapter*{#3}%
  \@mkboth{\MakeUppercase{#2}}{\MakeUppercase{#2}}%
}
\makeatother
```
但是这种解法导致章节间需要手动添加 `\clearpage` 换页，所以不如直接置该命令为空。

另见[水源社区的回答](https://shuiyuan.sjtu.edu.cn/t/%E6%B1%82%E6%95%99latex%E9%97%AE%E9%A2%98/112373/7)。
</details>
